macros/core/PGbasicmacros.pl

```diff
@@ -1182,8 +1182,24 @@ sub SOLUTION {
 	return "" if $solution_body eq "";
 
 	if ($displayMode =~ /HTML/) {
-		TEXT('<div class="knowl-container">',
-			knowlLink(SOLUTION_HEADING(), value => $solution_body, type => 'solution'), '</div>');
+		TEXT(tag(
+			'div',
+			class => 'solution accordion my-3',
+			tag(
+				'details',
+				class => 'accordion-item',
+				tag(
+					'summary',
+					class => 'accordion-button collapsed p-2',
```

Comment:
I think this looks a lot better if the horizontal margin is not changed.  That makes the alignment of the header text and the body text look off.  I think your primary goal was probably to reduce the vertical padding anyway.  You can do this by using `py-2` instead.  The same for hints as well of course.